spinalcordtoolbox/image.py

```diff
@@ -973,7 +973,17 @@ def get_dimension(im_file, verbose=1):
     ndims = [1, 1, 1, 1]
     pdims = [1, 1, 1, 1]
     data_shape = im_file.header.get_data_shape()
-    zooms = im_file.header.get_zooms()
+    # NB: Nibabel stores dim info as float32. Arithmetic using float32 values can cause issues. So, store the values
+    #     as rounded 64-bit float values, according to the spacing of the value.
+    zooms = []
+    for zoom_f32 in im_file.header.get_zooms():
+        # extract the exponent from the spacing used by the float32 values (i.e. 1e-9 -> '-9')
+        spacing = np.spacing(zoom_f32)
+        spacing_exp = np.floor(np.log10(np.abs(spacing))).astype(int)
+        # convert to python float, then round to remove any spurious digits
+        zoom_f64 = float(zoom_f32)
+        zoom_f64_rounded = round(zoom_f64, abs(spacing_exp) - 1)
+        zooms.append(zoom_f64_rounded)
```

Comment:
This change had a very tiny effect on the resampling test values, too:

```
_ test_sct_resample_output_has_expected_dimensions[dmri/dmri.nii.gz--f-0.5x0.5x1-expected_dim1] _

AssertionError: assert 1.682692 == 1.6826923 ± 1.0e-07
  
  comparison failed
  Obtained: 1.682692
  Expected: 1.6826923 ± 1.0e-07
```

I wonder if this old value had a spurious significant figure? I can double check to see exactly how it was originally computed.